### ISSUE TITLE
Option to split items for a queue in multiple sub-queues

### ISSUE THREAD

**Issue by gcastaldi** [swisspost/gateleen#550]
Sometimes queues created by the clients have a lot of items requiring time to be processed. It can be useful to add the option to split the incoming items in a set of sub-queues replacing the one specified by the client.
E.g.: given the queue 'myqueue' specified by the client the following queues are created in gateleen:
* myqueue-1
* myqueue-2
* myqueue-3
...
* myqueue-8

The number of sub-queues can be fixed in the configuration or variable considering an header (e.g. x-rp-deviceid).

**Comment by hiddenalpha** [comment]
@ljucam Isn't there a way to accomplish this somehow already? We somehow use this with
"*foo-bar-{x-vehicleid}*" where "*{x-vehicleid}*" gets replaced by the header in the incoming request. 

Eg an incoming request:
```
PUT /foo/bar HTTP/1.1
Host: example.com
x-vehicleid: 42
...
```

ends up in a queue named "*foo-bar-42*". **How does one configure this behavior?**

**Comment by mcweba** [comment]
@hiddenalpha You are right, there is the feature to define dynamic headers in the routing rules. We also use this feature. However, the proposed new feature has at least two advantages over the definition of headers in the rules:

1. With rule defined headers you need an aditional routing rule which creates an additional (internal) request. With the new feature the queue will be split before going to the routing rules and therefore create less load on the server.
2. With rule defined headers you need a suitable request header to be used for splitting. In our case we don't have something like a _x-vehicleId_ to split. We have a single backend which sends thousands of queue request to a single queue and there is no suitable request header that can be used

**Comment by Kusig** [comment]
The queue-split discriminator should be definable optionally as well in case of the URL eg. to use a path segment instead of just a fixed number of queues.

e.g. /some/api/{vehicleId}/some/endpoint

This may be possible by a configurable regex in the routing rules or, even better additionally, given as a hook property.

This way the order of messages could be guaranteed dependent on the well defined api uri without having to provide some kind of special header.

**Comment by gcastaldi** [comment]
> The queue-split discriminator should be definable optionally as well in case of the URL eg. to use a path segment instead of just a fixed number of queues.
> 
> e.g. /some/api/{vehicleId}/some/endpoint
> 
Yes, also this feature is implemented. It will be possible to use multiple path segments all added as postfixes.

**Comment by Kusig** [comment]
I think this issue could be closed, isn't it?

**Comment by mcweba** [comment]
Released in [v2.1.1](https://github.com/swisspost/gateleen/releases/tag/v2.1.1)
